ddcz/views/legacy.py

```diff
@@ -0,0 +1,135 @@
+import logging
+from django.apps import apps
+from ddcz.models.used.creations import Creation, CreativePage
+from django.http import HttpResponseRedirect
+from django.http.response import HttpResponse
+from django.shortcuts import get_object_or_404
+from django.urls.base import reverse
+from django.core.exceptions import ObjectDoesNotExist
+from django.views.decorators.http import require_http_methods
+
+# In the constant below: It is possible to vary the redirections.
+# It is preferable to do so via the constants rather than changing
+# the code if there is no real need for it, so the legacy router
+# remains automatized.
+
+CREATION = ["prispevky", "prispevky_komp"]
+CREATION_DETAIL = "prispevky_precti"
+
+# Forbidden dictionary keys for PAGE_TO_VIEW_MAP:
+#  - Anything from CREATION and CREATION_DETAIL
+#  - Anything from ALLOWED_CREATION_PAGES
+PAGE_TO_VIEW_MAP = {
+    "aktuality": "ddcz:news",
+    "novinky": "ddcz:newsfeed",
+    "uzivatele": "ddcz:users-list",
+    "registrace": "ddcz:sign-up",
+    "seznamka": "ddcz:dating",
+    "forum": "ddcz:phorum-list",
+    "linky": "ddcz:links-list",
+    "inzerce": "ddcz:market",
+    "credits": "ddcz:web-authors-and-editors",
+    "faq": "ddcz:website-manual",
+    "oao": "ddcz:faq",
+    "cojetodrd": "ddcz:about-drd",
+}
+
+COMMON_ARTICLES_NAME_MAP = {
+    "hrbitov": "hrbitov",
+    "clanky": "clanky",
+    "expanze": "expanze",
+    "kouzelnikzvl": "kouzelnikzvl",
+    "alchymistazvl": "alchymistazvl",
+    "valecnik": "valecnik",
+    "hranicar": "hranicar",
+    "zlodej": "zlodej",
+    "novapovolani": "novapovolani",
+    "noverasy": "noverasy",
+}
+
+
+ALLOWED_CREATION_PAGES = [
+    "bestiar",
+    "dobrodruzstvi",
+    "predmety",
+    "kouzla",
+    "alchpredmety",
+    "hranicarkouzla",
+    "dovednosti",
+    "galerie",
+    "fotogalerie",
+]
+
+
+@require_http_methods(["GET"])
+def legacy_router(request):
+
+    section = request.GET.get("rub", False)
+    subsection = request.GET.get("co", False)
+    id = request.GET.get("id", False)
+
+    # The LEGACY_PLAIN_ROUTER is redirecting basic pages.
+    # Typically no creative pages are present here.
+    if section in PAGE_TO_VIEW_MAP.keys():
+        return HttpResponseRedirect(reverse(PAGE_TO_VIEW_MAP[section]))
+
+    # Some of the creation have legacy url
+    # index.php?rub=prispevky(_komp)&co=page_slug
+    # Here are lists of such creative pages.
+    if section in CREATION and subsection in COMMON_ARTICLES_NAME_MAP.keys():
+        return HttpResponseRedirect(
+            reverse(
+                "ddcz:creation-list",
+                kwargs={"creative_page_slug": COMMON_ARTICLES_NAME_MAP[subsection]},
+            )
+        )
+
+    # For index.php?rub=prispevky_jeden&subsection=page_slug&id=article_id
```

Comment:
It's `co`, not `subsection`

Reply:
I renamed that, but in general this old terms do not seem to be very good systematic reference.

Reply:
I thought this is an example of an old URL and not a reference—if it would be, than `rub` should be `creative_page`, but IMHO that doesn't make sense.